transaction-view/src/bytes.rs

```diff
@@ -128,6 +145,51 @@ pub fn advance_offset_for_type<T: Sized>(bytes: &[u8], offset: &mut usize) -> Re
     Ok(())
 }
 
+/// Return a reference to the next slice of `T` in the buffer, checking bounds
+/// and advancing the offset.
+/// If the buffer is too short, return Err.
+///
+/// * `bytes` - Slice of bytes to read from.
+/// * `offset` - Curernt offset into `bytes`.
+/// * `num` - Number of `T` elements in the slice.
+///
+/// # Safety
```

Comment:
@alessandrod could you review the safety requirements here + the call-site documentation that we meet the constraints? (also for `read_type` below).